nowcasting_utils/models/loss.py

```diff
@@ -43,3 +51,310 @@ def get_mse_exp(self, output, target):
     def get_mae_exp(self, output, target):
         """Loss function weighted MAE"""
         return torch.sum(self.weights * torch.abs(output - target))
+
+
+class SSIMLoss(nn.Module):
+    def __init__(self, convert_range: bool = False, **kwargs):
+        """
+        SSIM Loss, optionally converting input range from [-1,1] to [0,1]
+        Args:
+            convert_range: Convert input from -1,1 to 0,1 range
+            **kwargs: Kwargs to pass through to SSIM
+        """
+        super(SSIMLoss, self).__init__()
+        self.convert_range = convert_range
+        self.ssim_module = SSIM(**kwargs)
+
+    def forward(self, x: torch.Tensor, y: torch.Tensor):
+        if self.convert_range:
+            x = torch.div(torch.add(x, 1), 2)
+            y = torch.div(torch.add(y, 1), 2)
+        return 1.0 - self.ssim_module(x, y)
```

Comment:
why is it 1 - here?

Reply:
Because SSIM has a score of 1 if its a perfect match, so this makes it minimize the value, with a loss of 0 if the target and output are identical

Reply:
ah nice, is it ok to add that in as a comment, then someone else reading it wont get confused